### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by manzanit0**
Added test file for Events endpoint.

### CODE REVIEW

File: src/tests/event.endpoint.test.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+require('../app.js');
+
+const assert = require('assert'),
+chai = require('chai'),
```
Comment (gil13) [inline src/tests/event.endpoint.test.js]: You should indent this requiring lines

File: src/tests/event.endpoint.test.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+chaiHttp = require('chai-http');
+
+const expect = chai.expect;
+chai.use(chaiHttp);
+
+const util = require('util');
```
Comment (gil13) [inline src/tests/event.endpoint.test.js]: Unused require

File: src/tests/event.endpoint.test.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+describe('Event API endpoint', () => {
+
+    describe('GET all events', () => {
+        it('GET all event should have a 200 status', (done) => { // <= Pass in done callback
+            chai.request('http://localhost:3000')
```
Comment (gil13) [inline src/tests/event.endpoint.test.js]: You can put the url in a external variable

File: src/tests/event.endpoint.test.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    done();
+            });
+        });
+
+    });
+
+    describe('PUT event by Id', () => {
+        it('PUT event receives an 201 status', (done) => {
+            chai.request('http://localhost:3000')
+                .post('/event')
```
Comment (gil13) [inline src/tests/event.endpoint.test.js]: You say this is a put test but you write a post test

File: src/tests/event.endpoint.test.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                .send({id: 987654321, title: 'ola k ase 2', descripcion: '', fecha: '01/11/1234'})
+                .end((err, res) => {
+                    expect(res).to.have.status(201);
+                    done();
+            });
+        });
+
+        it('PUT event updates the object and its not equals', (done) => {
+            const sentData = {id: 12345, title: 'ola k ase 2', descripcion: '', fecha: '01/11/1234'};
```
Comment (gil13) [inline src/tests/event.endpoint.test.js:112]: This data insert could be placed in a beforeEach hook

File: src/endpoints/event.endpoint.js [hunk 6]
```diff
@@ -121,7 +121,7 @@ const deleteEvent = (req, res) => {
         for (let i = 0; events.length > i; i++) {
```
Comment (arielcamus) [inline src/endpoints/event.endpoint.js:121]: Could you use a dictionary where the key is the courseId instead? You wouldn't have to loop through the array.

File: src/tests/event.endpoint.test.js (hunk 1 above)
Comment (manzanit0) [reply src/tests/event.endpoint.test.js]: @gil13 ¿marca alguna diferencia encadenarlas con coma o meter cada require en una variable?

File: src/tests/event.endpoint.test.js (hunk 5 above)
Comment (manzanit0) [reply src/tests/event.endpoint.test.js:112]: @gil13 He estado mirando como funcionan los beforeEach hooks... ¿Realmente nos interesa usarlos para nuestro caso? Quiero decir, supuestamente se ejecuta el hook antes de cada "it", de modo que estaríamos insertando antes de cada post/get/put, mientras que sólo nos interesa en el put, ¿no?

**Comment by manzanit0** [comment]
@gil13 echa un vistazo al fichero. He corregido algunas cosas y he planteado otras. Hay un puñado que no estoy seguro cómo se harían realmente, o si ni siquiera se hacen en JS (inserción previa y borrado al final?).

File: src/endpoints/event.endpoint.js (hunk 6 above)
Comment (manzanit0) [reply src/endpoints/event.endpoint.js:121]: @arielcamus Hm, yes! I didn't know JS had dictionaries lmao

Nonetheless,now that we've implemented all this logic with Mongo, we just use a findByIdAndRemove, or something like that, so it's pretty straight forward :)

File: src/tests/event.endpoint.test.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
+require('../app.js');
+
+const assert = require('assert'),
+        chai = require('chai'),
+    chaiHttp = require('chai-http'),
+        util = require('util'); // What's the point of Util?
```
Comment (gootyfer) [inline src/tests/event.endpoint.test.js:6]: Unused import

File: src/tests/event.endpoint.test.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        //TODO: Inserción de datos de test en la BD.
+    });
+
+    after( () => {
+        //TODO: borrado de los datos de test en la BD.
+    });
+
+    describe('GET all events', () => {
+        it('GET all event should have a 200 status', (done) => { // <= Pass in done callback
+            chai.request('serverUrl')
```
Comment (gootyfer) [inline src/tests/event.endpoint.test.js:26]: You should get the server URL from the variable, not a hardcoded string.

File: src/tests/event.endpoint.test.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+    });
+
+    describe('PUT event by Id', () => {
+        it('PUT event receives an 201 status', (done) => {
+            chai.request(serverUrl)
+                .put('/event/12345')
+                .send({id: 987654321, title: 'ola k ase 2', descripcion: '', fecha: '01/11/1234'})
+                .end((err, res) => {
+                    expect(res).to.have.status(201);
```
Comment (gootyfer) [inline src/tests/event.endpoint.test.js:106]: It seems you get a 200 status here. Maybe you should check it begins with '2' to make sure it is OK.

**Review by gootyfer — COMMENTED** [review]
Nice first approach to the testing milestone. But I'd test not only the status code of the response but also that the data is added/update/deleted from the in-memory array. Great work, guys!

**Pull request merged**